File: app/src/main/java/net/opendasharchive/openarchive/onboarding/CustomOnboardingScreen.java
package net.opendasharchive.openarchive.onboarding;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import net.opendasharchive.openarchive.R;

public class CustomOnboardingScreen extends Fragment {

    private static final String ARG_LAYOUT_RES_ID = "layoutResId";
    private static final String ARG_LAYOUT_TITLE = "layoutTitle";
    private static final String ARG_LAYOUT_SUBTITLE = "layoutSubtitle";
    private static final String ARG_LAYOUT_IMAGE = "layoutImage";

    private int layoutResId;
    private String mTitle;
    private String mSubTitle;
    private int mImageResource;

    private Button mButton;

    public static CustomOnboardingScreen newInstance(int layoutResId, String title, String subtitle, int image) {
        CustomOnboardingScreen sampleSlide = new CustomOnboardingScreen();

        Bundle args = new Bundle();
        args.putInt(ARG_LAYOUT_RES_ID, layoutResId);
        args.putString(ARG_LAYOUT_TITLE,title);
        args.putString(ARG_LAYOUT_SUBTITLE,subtitle);
        args.putInt(ARG_LAYOUT_IMAGE,image);
        sampleSlide.setArguments(args);

        return sampleSlide;
    }

    public void setTitle (String title)
    {
        mTitle = title;
    }

    public void setSubTitle(String subTitle) { mSubTitle = subTitle; }

    public void setImageResource (int resource) { mImageResource = resource; }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getArguments() != null && getArguments().containsKey(ARG_LAYOUT_RES_ID)) {
            layoutResId = getArguments().getInt(ARG_LAYOUT_RES_ID);
        }

        if (getArguments() != null && getArguments().containsKey(ARG_LAYOUT_TITLE)) {
            mTitle = getArguments().getString(ARG_LAYOUT_TITLE);
        }

        if (getArguments() != null && getArguments().containsKey(ARG_LAYOUT_SUBTITLE)) {
            mSubTitle = getArguments().getString(ARG_LAYOUT_SUBTITLE);
        }

        if (getArguments() != null && getArguments().containsKey(ARG_LAYOUT_IMAGE)) {
            mImageResource = getArguments().getInt(ARG_LAYOUT_IMAGE);
        }


    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(layoutResId, container, false);
        ((TextView)view.findViewById(R.id.custom_slide_big_text)).setText(mTitle);

        if (!TextUtils.isEmpty(mSubTitle)) {

            TextView tv =
                    (TextView)view.findViewById(R.id.custom_slide_big_text_sub);
            tv.setText(mSubTitle);
            tv.setVisibility(View.VISIBLE);
        }

        ((ImageView)view.findViewById(R.id.custom_slide_image)).setImageResource(mImageResource);

        mButton = view.findViewById(R.id.custom_slide_button);
        if (mButton != null && mButtonListener != null) {
            mButton.setText(mButtonText);
            mButton.setVisibility(View.VISIBLE);
            mButton.setOnClickListener(mButtonListener);
        }
        return view;

    }

    private String mButtonText;
    private View.OnClickListener mButtonListener;

    public void enableButton (String buttonText, View.OnClickListener listener)
    {
        mButtonText = buttonText;
        mButtonListener = listener;
    }
}

File: app/src/main/java/net/opendasharchive/openarchive/features/onboarding/CustomOnboardingScreen.kt
package net.opendasharchive.openarchive.features.onboarding

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import net.opendasharchive.openarchive.databinding.CustomOnboardingMainBinding
import net.opendasharchive.openarchive.util.extensions.show

class CustomOnboardingScreen : Fragment() {

    private var mTitle = 0
    private var mSubTitle = 0
    private var mImageResource = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        mTitle = arguments?.getInt(ARG_LAYOUT_TITLE) ?: 0
        mSubTitle = arguments?.getInt(ARG_LAYOUT_SUBTITLE) ?: 0
        mImageResource = arguments?.getInt(ARG_LAYOUT_IMAGE) ?: 0
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        val binding = CustomOnboardingMainBinding.inflate(inflater, container, false)

        binding.customSlideBigText.setText(mTitle)

        if (mSubTitle != 0) {
            binding.customSlideBigTextSub.setText(mSubTitle)
            binding.customSlideBigTextSub.show()
        }

        binding.customSlideImage.setImageResource(mImageResource)

        return binding.root
    }

    companion object {

        private const val ARG_LAYOUT_TITLE = "layoutTitle"
        private const val ARG_LAYOUT_SUBTITLE = "layoutSubtitle"
        private const val ARG_LAYOUT_IMAGE = "layoutImage"

        fun newInstance(title: Int, subtitle: Int, image: Int): CustomOnboardingScreen
        = CustomOnboardingScreen().also {
            it.arguments = Bundle().apply {
                putInt(ARG_LAYOUT_TITLE, title)
                putInt(ARG_LAYOUT_SUBTITLE, subtitle)
                putInt(ARG_LAYOUT_IMAGE, image)
            }
        }
    }
}